### PR TITLE
Support all unsigned int sizes on unique

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123643

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Sounds good!

**Review by kit1980 — APPROVED** [review]

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change, but it is failing lots of jobs with the new dtype https://hud.pytorch.org/pytorch/pytorch/commit/8aa08b8b9d1fab2a13dc5fbda74c553cb2a08729' -c weird

I don't see pull/trunk jobs running on the PR https://hud.pytorch.org/pr/123643, so the signals were missed.  This was caused by an incidence we had yesterday where these workflows were accidentally disabled for about 2 hours.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -16832,8 +16832,8 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            skips=(
            )),
     OpInfo('unique',
-           dtypes=all_types_and(torch.bool, torch.float16, torch.bfloat16),
-           dtypesIfCUDA=all_types_and(torch.bool, torch.float16),
+           dtypes=all_types_and(torch.bool, torch.float16, torch.bfloat16, torch.uint16, torch.uint32, torch.uint64),
+           dtypesIfCUDA=all_types_and(torch.bool, torch.float16, torch.uint16, torch.uint32, torch.uint64),
```
Comment (sshleifer) [inline torch/testing/_internal/common_methods_invocations.py:16836]: torch.uint16 exists?

Comment (ezyang) [reply torch/testing/_internal/common_methods_invocations.py:16836]: yeah! It doesn't support that many ops but we're happy to add them as requested

Comment (ezyang) [reply torch/testing/_internal/common_methods_invocations.py:16836]: It's mostly useful with torch.compile, which has fairly complete compilation coverage for it.

**Pull request closed without merging**